src/solvers/flattening/boolbv.cpp

```diff
@@ -29,15 +29,13 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include "../floatbv/float_utils.h"
 
-bool boolbvt::literal(
-  const exprt &expr,
-  const std::size_t bit,
-  literalt &dest) const
+optionalt<literalt>
```

Comment:
I don't see why this change is needed.  If there is a real error then an exception should be thrown.  If not then returning constant literals is acceptable.

Reply:
The function is not returning a constant literal when there is an error, it is just leaving `dest` untouched. So this is not changing the behavior of the function, just making the interface clearer.

Reply:
May I suggest a different route: please just `#ifdef 0` this code. Whether you change the interface to an `optionalt` I don't care as much then. (Background: I wanted to figure out whether throwing an exception would be fine. I couldn't tell, because the code is never used.)

Reply:
Not sure I get your comment. If I `#ifdef 0` the function when cbmc will not build because `get_bool` is called from somewhere.

Reply:
Sorry, this is not about `get_bool`, but about `literal` (both `boolbvt::literal` and `prop_convt::literal`).